src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,488 @@
+		'wp_terms_list_table'                         => 'wp-admin/includes/class-wp-terms-list-table.php',
+		'wp_theme_install_list_table'                 => 'wp-admin/includes/class-wp-theme-install-list-table.php',
+		'wp_themes_list_table'                        => 'wp-admin/includes/class-wp-themes-list-table.php',
+		'wp_upgrader_skin'                            => 'wp-admin/includes/class-wp-upgrader-skin.php',
+		'wp_upgrader'                                 => 'wp-admin/includes/class-wp-upgrader.php',
+		'wp_users_list_table'                         => 'wp-admin/includes/class-wp-users-list-table.php',
+		'wp_user_search'                              => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_export_requests_table'       => 'wp-admin/includes/deprecated.php',
+		'wp_privacy_data_removal_requests_table'      => 'wp-admin/includes/deprecated.php',
+	);
+
+	/**
+	 * Whether the autoloader has already been registered or not.
+	 *
+	 * Avoid registering the autoloader multiple times.
+	 *
+	 * @static
+	 * @access private
+	 *
+	 * @var bool
+	 */
+	private static $registered = false;
+
+	/**
+	 * Register the autoloader.
+	 *
+	 * @return void
+	 */
+	public static function register() {
+		// Bail early if already registered.
+		if ( static::$registered ) {
+			return;
+		}
+
+		static::register_external_bundled();
+		static::register_core();
+
+		static::$registered = true;
+	}
+
+	/**
+	 * Register the autoloader for external, bundled libraries.
+	 *
+	 * @return void
+	 */
+	public static function register_external_bundled() {
+		require_once ABSPATH . 'wp-includes/Requests/src/Autoload.php';
+		require_once ABSPATH . 'wp-includes/sodium_compat/autoload.php';
+
+		spl_autoload_register( array( '\WpOrg\Requests\Autoload', 'load' ) );
+	}
+
+	/**
+	 * Register the autoloader for WordPress Core classes.
+	 *
+	 * @return void
+	 */
+	public static function register_core() {
+		spl_autoload_register( array( __CLASS__, 'autoload_core' ), true, true );
+	}
+
+	/**
+	 * Autoload a WordPress class.
+	 *
+	 * @param string $class Class name.
+	 * @return bool True if the class was loaded, false otherwise.
+	 */
+	public static function autoload_core( $class_name ) {
+		// Lowercase the classname to accomodate for WP classes written with wrong cases.
+		$class_name = strtolower( $class_name );
+
+		// Bail early if the class is not a WP class.
+		if ( ! array_key_exists( $class_name, static::CLASSES_PATHS ) ) {
+			return false;
+		}
+
+		/*
+		 * SimplePie classes should additionally load the `wp-includes/class-simplepie.php` file
+		 * to ensure that constants are defined.
+		 */
+		if ( 0 === strpos( $class_name, 'simplepie' ) ) {
+			require_once ABSPATH . static::CLASSES_PATHS['simplepie'];
+		}
+
+		require_once ABSPATH . static::CLASSES_PATHS[ $class_name ];
+		return true;
+	}
+}
+
+// Register the autoloader.
+WP_Autoload::register();
```

Comment:
Would it be better to move these lines to `src/index.php` after `require_once ABSPATH . WPINC . '/class-wp-autoload.php';`? That way only the class is being defined in the file. AFAIK that is better form.

Reply:
Ah, but this is being required in multiple places. So maybe not.

Reply:
Yeah, there are multiple entry points in WP, so I needed to add it in more places than one 👍